docs/block-validation/block-format.md

```diff
@@ -20,5 +20,54 @@ The FullBlock has fields for both the trunk and the foliage of the blockchain. T
 - **foliage**: Foliage: Foliage data for the reward chain block, the hash of this is the `header_hash`.
 - **foliage_transaction_block**: Optional[FoliageTransactionBlock]: Transaction related metadata that is relevant for light clients (not actual transactions), only for tx blocks.
 - **transactions_info**: Optional[TransactionsInfo]: Transaction related metadata that is not relevant for light clients (not actual transactions), only for tx blocks.
-- **transactions_generator**: Optional[SerializedProgram]: A clvm program that generates all transactions (spends).
+- **transactions_generator**: Optional[SerializedProgram]: A clvm/rust program that generates all transactions (spends). See the next section for an important [update](#transactions_generator-update) due to the 2.1.0 hard fork.
 - **transactions_generator_ref_list**: List[uint32]: A list of block heights of previous generators referenced by this block's generator.
+
+## transactions_generator update
+
+Chia underwent a hard fork in version 2.1.0. This included updates to the `transactions_generator` code.
+
+### When
+
+The hard fork will activate at block `5 496 000`, which is expected to occur in June 2024. All wallets need to be compatible with the new implementation prior to this block.
```

Comment:
technically, there's no change to the wallet protocol. I think this should probably say "nodes" instead of "wallets"

Reply:
If they use the transaction generator to calculate cost it would affect them as well, I believe.